spec/lib/checking-account_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'spec_helper'
+
+describe BankAccounts::CheckingAccount do
+  describe "class methods" do
+    it "responds to 'new'" do
+      expect(BankAccounts::CheckingAccount).to respond_to :new
+    end
+
+    describe "inherits methods from Account class" do
+      let(:checking_account) do
+        BankAccounts::CheckingAccount.new("Lindsay", 10_000)
+      end
+
+        it "adds deposit(amount) and returns updated balance" do
+          expect(checking_account.deposit(1_010)).to eq(11_010)
+        end
+
+      describe "raise_exception when initial_balance is nil" do
+        it "raises an ArgumentError" do
+          expect { BankAccounts::CheckingAccount.new(898776, nil) }.to raise_exception(ArgumentError)
+        end
+      end
+    end
+
+    context "when withdraw(amount) would result in a negative value of balance" do
+      let(:checking_account) do
+        BankAccounts::CheckingAccount.new("Voltron2", 1002)
+      end
+
+        it "returns original balance" do
+          expect(checking_account.withdraw(5032)).to eq(1002)
+        end
+
+        it "allows withdrawals betwee 0 and 10" do
+          expect(checking_account.withdraw(1000)).to eq(1)
```

Comment:
What is this spec supposed to test? The description is not clear. It says something about withdrawing 0 and 10, but the test is withdrawing 1000?